spec/index.bs

```diff
@@ -1113,6 +1101,20 @@ Given an {{HTMLScriptElement}} (|script|), this algorithm performs the following
 
     If the algorithm threw an error, rethrow the error.
 
+## <dfn-abstract-op export>Validate attribute mutation</dfn> ## {#validate-attribute-mutation}
+Given {{Attr}} |attribute|, {{Element}} |element| and {{TrustedType}} or a string |newValue|, this algorithm performs the following steps:
+
+1. If |element| has an <a spec="webidl" lt="attribute">IDL attribute</a> that [=reflects=] a content attribute with |attribute|'s [=Attr/local name=], let |idlAttribute| be that IDL attribute. Otherwise, return stringified |newValue|.
+1. If {{StringContext}} IDL extended attribute does not appears on |idlAttribute|, return stringified |newValue|.
+1. Set |value| to the result of executing [$Get Trusted Type compliant string$] with the following arguments:
+     * {{StringContext}} identifier as |expectedType|
+     * |newValue| as |input|
+     * |element|'s <a>relevant global object</a> as |global|
```

Comment:
Do we want this or the element's node document's global? Is there test coverage for this? (Gecko mutates the global when moving elements across documents which is against the specification.) I think in another case we decided to use the element's node document for cases like this, not because of Gecko, but because it would be more likely what is expected.

cc @domenic

Reply:
I think it's fine to use the global from the node's document instead, but can you give me an example of when those two globals differ when this algorithm is called? Is it only in Gecko? I don't think we have tests currently for moving nodes across documents and calling setAttribute(), this would help us add a case for this.

Reply:
When you move a node from another document its own global will be that other document's global (except in Gecko), but it's node document's global will be that of the new document its been inserted into.

Reply:
> Do we want this or the element's node document's global? Is there test coverage for this? (Gecko mutates the global when moving elements across documents which is against the specification.) I think in another case we decided to use the element's node document for cases like this, not because of Gecko, but because it would be more likely what is expected.
> 
> cc @domenic

@annevk Expected by or for what?

Reply:
By web developers.